src/components/tabs/tabs.ts

```diff
@@ -181,17 +205,28 @@ export class MdTabGroup {
     }
   }
 
-  /** Increment the focus index by 1; prevent going over the number of tabs */
+  /** Increment the focus index by 1 until a valid tab is found. */
   focusNextTab(): void {
-    if (this._labelWrappers && this.focusIndex < this._labelWrappers.length - 1) {
-      this.focusIndex++;
+    if (this._labelWrappers) {
+      const tabs: MdTab[] = this._tabs.toArray();
+      for (let i = this.focusIndex + 1; i < tabs.length; i++) {
+        if (this.isValidIndex(i)) {
+          this.focusIndex = i;
+          return;
+        }
+      }
     }
   }
 
-  /** Decrement the focus index by 1; prevent going below 0 */
+  /** Decrement the focus index by 1 until a valid tab is found. */
   focusPreviousTab(): void {
-    if (this.focusIndex > 0) {
-      this.focusIndex--;
+    if (this._labelWrappers) {
+      for (let i = this.focusIndex - 1; i >= 0; i--) {
+        if (this.isValidIndex(i)) {
+          this.focusIndex = i;
+          return;
+        }
+      }
```

Comment:
Lots of duplicated code between `focusPreviousTab` and `focusNextTab`. You should do something like
```ts
focusNextTab() {
  moveFocus(1);
}

focusPreviousTab() {
  moveFocus(-1);
}

moveFocus() {
  // ...
}
```